File: java/dtdgen-master/src/main/java/com/philhanna/dtdgen/ChildModel.java
package com.philhanna.dtdgen;

/**
 * Records information about the presence of a child element within its
 * parent element. If the parent element is sequenced, then the child
 * elements always occur in sequence with the given frequency.
 */
public class ChildModel {
   
   // ==================================================================
   // Instance variables
   // ==================================================================
   
   private final String name;
   private boolean repeatable;
   private boolean optional;

   // ==================================================================
   // Constructors
   // ==================================================================

   /**
    * Creates a ChildModel of the specified child element name
    * @param name the child element name
    */
   public ChildModel(String name) {
      this.name = name;
   }

   // ==================================================================
   // Implementation of ChildModel interface
   // ==================================================================
   
   /**
    * Returns the child element name
    */
   public String getName() {
      return name;
   }

   /**
    * Returns <code>true</code> if this child can be repeated in its
    * parent element
    */
   public boolean isRepeatable() {
      return repeatable;
   }

   /**
    * Returns true if this child element is not required to be present
    * under its parent element
    */
   public boolean isOptional() {
      return optional;
   }

   /**
    * Sets the repeatable attribute
    * @param repeatable the repeatable attribute
    */
   public void setRepeatable(boolean repeatable) {
      this.repeatable = repeatable;
   }

   /**
    * Sets the optional attribute
    * @param optional the optional attribute
    */
   public void setOptional(boolean optional) {
      this.optional = optional;
   }
}

File: dtdgen/child_model.py
class ChildModel:
    """Records information about the presence of a child element within its
    parent element. If the parent element is sequenced, then the child
    elements always occur in sequence with the given frequency."""

    def __init__(self, name: str):
        """Creates a ChildModel of the specified child element name"""
        self._name: str = name
        self._repeatable: bool = False
        self._optional: bool = True

    @property
    def name(self):
        return self._name

    @property
    def repeatable(self):
        return self._repeatable

    @repeatable.setter
    def repeatable(self, value: bool):
        self._repeatable = value

    @property
    def optional(self):
        return self._optional

    @optional.setter
    def optional(self, value: bool):
        self._optional = value

    def __str__(self):
        class_name = self.__class__.__name__
        return (
            f"{class_name} [name={self.name}"
            f", repeatable={self.repeatable}"
            f", optional={self.optional}"
            "]"
        )